### PR TITLE
quote project dir when used in regex

### PR DESCRIPTION
**Pull request by jgeraerts**
Before submitting a PR make sure the following things have been done:

- [ ] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests to cover your change(s)
- [ ] All tests are passing
- [ ] The new code is not generating reflection warnings
- [ ] You've updated the readme (if adding/changing middleware)

Thanks!

Since windows uses backslashes this results in an invalid regular
expression throwing a PatternSyntaxException. By passing the
project-root through Pattern/quote it is escaped properly for use in
a regular expression

### CODE REVIEW

File: src/cider/nrepl/middleware/util/namespace.clj [hunk 1]
```diff
@@ -24,11 +25,15 @@
   (str (System/getProperty "user.dir")
        (System/getProperty "file.separator")))
 
+(defn- re-quote
```
Comment (Malabarba) [inline src/cider/nrepl/middleware/util/namespace.clj]: I think there's no need for this function. We can just use Pattern/quote directly.

Comment (bbatsov) [inline src/cider/nrepl/middleware/util/namespace.clj]: Indeed.

**Comment by bbatsov** [comment]
You should capitalise the first word in the commit's title (quote -> Quote). In the commit's description the final sentence is missing its `.`.

**Comment by jgeraerts** [comment]
Commit updated with requests.

**Comment by jgeraerts** [comment]
I was also wondering if I could use `#(String/startsWith (str %) project-root)` instead of the whole regex setup.

**Comment by bbatsov** [comment]
> I was also wondering if I could use #(String/startsWith (str %) project-root) instead of the whole regex setup.

`String` has a static method `startsWith`? In general yeah - the simpler the solution the better.

**Comment by jgeraerts** [comment]
no sorry it's wrong. it should be `(.startsWith .... )`.

**Comment by bbatsov** [comment]
As mentioned above - I believe this simpler approach will work just fine, so feel free to update the PR.

**Comment by jgeraerts** [comment]
i've replace the code with `(.startsWith ... )` construct.

**Comment by bbatsov** [comment]
One final remark - now the title of the commit doesn't correspond with the actual change.

**Comment by jgeraerts** [comment]
done

**Comment by bbatsov** [comment]
:+1: Thanks!

**Comment by Malabarba** [comment]
Thanks!

**Pull request merged**